vmdk_plugin/drivers/vmdk/vmdk_driver.go

```diff
@@ -298,50 +296,81 @@ func (d *VolumeDriver) processMount(r volume.MountRequest) volume.Response {
 	return volume.Response{Mountpoint: mountpoint}
 }
 
-// No need to actually manifest the volume on the filesystem yet
-// (until Mount is called).
-// Name and driver specific options passed through to the ESX host
-
-// Create - create a volume.
-func (d *VolumeDriver) Create(r volume.Request) volume.Response {
-
+// prepareCreateOptions sets default options for create request r.
+func (d *VolumeDriver) prepareCreateOptions(r volume.Request) error {
 	if r.Options == nil {
 		r.Options = make(map[string]string)
 	}
-	// If cloning a existent volume, create and return
-	if _, result := r.Options["clone-from"]; result == true {
-		errClone := d.ops.Create(r.Name, r.Options)
-		if errClone != nil {
-			log.WithFields(log.Fields{"name": r.Name, "error": errClone}).Error("Clone volume failed ")
-			return volume.Response{Err: errClone.Error()}
-		}
-		return volume.Response{Err: ""}
-	}
 
-	// Use default fstype if not specified
-	if _, result := r.Options["fstype"]; result == false {
+	// Use default fstype if both fstype and clone-from are not specified.
+	_, fstypeRes := r.Options["fstype"]
+	_, cloneFromRes := r.Options["clone-from"]
+	if !fstypeRes && !cloneFromRes {
+		log.WithFields(log.Fields{"req": r}).Debugf("Setting fstype to %s ", fs.FstypeDefault)
```

Comment:
There's an explicit check in [`vmdk_ops.py`](https://github.com/vmware/docker-volume-vsphere/blob/b93c186ad2a05c35223d396a4629f74706fb6287/esx_service/vmdk_ops.py#L491) that raises an error on seeing `fstype` in a clone-from request. Therefore, this separation is not directly possible.